Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132327034)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs.meta	(date 1763132327034)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 2094347624394ff889c33e18dbbfdbc3
+timeCreated: 1763132327
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132330272)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs	(date 1763132330272)
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Atomic.Entities
+{
+    public interface IFixedTickEntitySystem<E> : IEntitySystem<E> where E : IEntity
+    {
+        void FixedTick(ICollection<E> collection, float deltaTime);
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132330266)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IFixedTickEntitySystem.cs.meta	(date 1763132330266)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 53ac2da2579f449388e0a00c4e5a6a0b
+timeCreated: 1763132330
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132333709)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs	(date 1763132333709)
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Atomic.Entities
+{
+    public interface ILateTickEntitySystem<E> : IEntitySystem<E> where E : IEntity
+    {
+        void LateTick(ICollection<E> collection, float deltaTime);
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132319446)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs.meta	(date 1763132319446)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 027f61d481ec455f9090f38d07048ee1
+timeCreated: 1763132319
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132323065)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs	(date 1763132323065)
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Atomic.Entities
+{
+    public interface IDisableEntitySystem<E> : IEntitySystem<E> where E : IEntity
+    {
+        void Disable(ICollection<E> collection);
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132323059)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IDisableEntitySystem.cs.meta	(date 1763132323059)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 46d88676739c463d8f7dd9c408c5b2dc
+timeCreated: 1763132323
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132327041)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ITickEntitySystem.cs	(date 1763132327041)
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Atomic.Entities
+{
+    public interface ITickEntitySystem<E> : IEntitySystem<E> where E : IEntity
+    {
+        void Tick(ICollection<E> collection, float deltaTime);
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems.meta
new file mode 100644
--- /dev/null	(date 1763132175806)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems.meta	(date 1763132175806)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: a29d1517a6124c5987a6029f3fbba5bb
+timeCreated: 1763132175
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132336044)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs	(date 1763132336044)
@@ -0,0 +1,6 @@
+namespace Atomic.Entities
+{
+    public interface IEntitySystem<E> where E : IEntity
+    {
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132187511)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEntitySystem.cs.meta	(date 1763132187511)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 57190b27e55e44e8a9fdc7f230662e16
+timeCreated: 1763132187
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs
new file mode 100644
--- /dev/null	(date 1763132319453)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/IEnableEntitySystem.cs	(date 1763132319453)
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Atomic.Entities
+{
+    public interface IEnableEntitySystem<E> : IEntitySystem<E> where E : IEntity
+    {
+        void Enable(ICollection<E> collection);
+    }
+}
\ No newline at end of file
Index: Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs.meta
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs.meta b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs.meta
new file mode 100644
--- /dev/null	(date 1763132333701)
+++ b/Assets/Plugins/Atomic/Entities/Scripts/Systems/ILateTickEntitySystem.cs.meta	(date 1763132333701)
@@ -0,0 +1,3 @@
+fileFormatVersion: 2
+guid: 2c7f38911cb649818d3519f550834e42
+timeCreated: 1763132333
\ No newline at end of file
